cryostatio/cryostat3
# feat(auth): remove integrated auth, delegate to proxy

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Related to #208
Depends on https://github.com/cryostatio/cryostat-web/pull/1181

## How to manually test:
1. Check out `src/main/webui` to -web PR above, then build cryostat3 container
2. `smoketest.bash -OXgtb`
3. After some time, `http://localhost:8080` (note `8080`, not `8181`) should open in your browser. Use `user:pass` to log in. This is the new authenticating reverse proxy setup which handles authentication/authorization before the user reaches the actual application. The Cryostat server is still listening on port 8181, but you should not be able to reach that anymore - it is bound to `localhost` only and not exposed from the compose network to your host machine. You can also do ex. `http --auth=user:pass :8080/api/v3/targets` the same way you would previously have done `http --auth=user:pass :8181/api/v3/targets`. In any case, this change should be more or less transparent, other than the login screen looking different.
